File: missions/30_stdout_stderr_redirection/merlin.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <time.h>

char msg[] = "LACLESECRETEESTSURSTDERR";
char cle[] = "MRGLPZYMAMLAJNBWRQPYACQBDYUFWQFUWVXQELCCGCJZUEJUEZ"
             "XLDYQPVHLHGLXKZJAAUKKFCTQJQKMMCNLUUEFFYEYPHNWCMKDM"
             "BKQAJPSHGBTJJQKDAAPCWFHTKGLTWRFTEWMPDHCHQNVVLHDWNM"
             "LFCLEAZWCQLVBXFFRRZMMJGLMKEVUXXDHUCHWAYQJFWAUYEHUP";

int main(int argc, char** argv)
{
    if (argc > 1) {
        printf("%s takes no argements\n", argv[0]);
        return 1;
    }
    int len_msg = strlen(msg);
    int len_cle = strlen(cle);

    int pos_msg = 0;
    int pos_cle = 0;

    srand(time(NULL));

    while (pos_msg < len_msg || pos_cle < len_cle) {
        if (rand() % 6 == 0 && pos_msg < len_msg) {
            fprintf(stdout, "%c", msg[pos_msg++]);
            fflush(stdout);
        } else {
            fprintf(stderr, "%c", cle[pos_cle++]);
            fflush(stderr);
        }
    }

    fprintf(stdout, "\n");
    fprintf(stderr, "\n");

    return 0;
}

File: missions/30_stdout_stderr_redirection/merlin.py
#!/usr/bin/env python3

msg = "LACLESECRETEESTSURSTDERR"
cle = ("MRGLPZYMAMLAJNBWRQPYACQBDYUFWQFUWVXQELCCGCJZUEJUEZ"
       "XLDYQPVHLHGLXKZJAAUKKFCTQJQKMMCNLUUEFFYEYPHNWCMKDM"
       "BKQAJPSHGBTJJQKDAAPCWFHTKGLTWRFTEWMPDHCHQNVVLHDWNM"
       "LFCLEAZWCQLVBXFFRRZMMJGLMKEVUXXDHUCHWAYQJFWAUYEHUP")

if __name__ == "__main__":
    from sys import argv, exit, stderr
    from random import randint

    if len(argv) > 1:
        print(f"L'executable {argv[0]} ne prend pas d'arguments.")
        exit(1)

    i, j = 0, 0
    while i < len(msg) or j < len(cle):

        if j >= len(cle) or (randint(0, len(msg)+len(cle)) < len(msg) and i < len(msg)):
            print(msg[i], end="", flush=True)
            i += 1
        else:
            print(cle[j], end="", file=stderr, flush=True)
            j += 1
    print("")
    print("", file=stderr)
